### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by AlexeyBavika**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -8,7 +13,30 @@
...
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder wordsStartWithW = new StringBuilder();
+        try (Scanner scanner = new Scanner(file)) {
+            if (!scanner.hasNext()) {
+                return new String[0];
+            }
+            while (scanner.hasNext()) {
+                String word = scanner.next();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:27]:
```suggestion
                String word = scanner.next().toLowerCase();
```

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/FileWork.java:27]: and then you don't have to check for Bigger W

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -8,7 +13,30 @@
...
+                if (word.startsWith(BIG_W) || word.startsWith(SMALL_W)) {
+                    wordsStartWithW.append(word.toLowerCase().replaceAll("[^a-zA-z]", "") + " ");
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        String[] sortedWords = wordsStartWithW.toString().trim().split(" ");
+        if (sortedWords.length <= 1) {
+            return new String[0];
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:37]: do we need this check?

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -8,7 +13,26 @@
...
+                return new String[0];
+            }
+            while (scanner.hasNext()) {
+                String word = scanner.next().toLowerCase();
+                if (word.startsWith(SMALL_W)) {
+                    wordsStartWithW.append(word.replaceAll("[^a-zA-Z]", "") + " ");
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:32]: you should re-throw exception, not ignore them

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -8,7 +13,26 @@
...
+        }
+        String[] sortedWords = wordsStartWithW.toString().trim().split(" ");
+        Arrays.sort(sortedWords);
+        return sortedWords.length > 1 ? sortedWords : new String[0];
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:36]: and still, do we need this check?

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -8,7 +13,26 @@
...
+    private static final String SMALL_W = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder wordsStartWithW = new StringBuilder();
+        try (Scanner scanner = new Scanner(file)) {
+            if (!scanner.hasNext()) {
+                return new String[0];
+            }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:24]: do we need this check?

**Comment by AlexeyBavika** [comment]
unfortunately its impossible to remove this check because at the end method split adds space and length = 1 then.

**Pull request closed without merging**